src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -212,7 +212,12 @@ def remap(
     __call__ = remap  # type: ignore[assignment]
 
     def restrict(self, index: common.AnyIndexSpec) -> common.Field:
-        new_domain, buffer_slice = self._slice(index)
+        new_index: Tuple | common.AnyIndexSpec = index
+        if isinstance(index, slice):
+            new_index = self._refactor_slice(index)
+        if isinstance(index, tuple) and all(isinstance(i, slice) for i in index):
+            new_index = tuple([self._refactor_slice(i) for i in index])  # type: ignore[arg-type] # all i's are slices as per if statement
```

Comment:
Let's wrap this in a utility function, e.g. `canonicalize_any_index_sequence`, then we can unit testing is easier.
Probably the canonicalization can be called from `_slice` after `index_sequence = common.as_any_index_sequence(index)`.

The unit test would take the new form and you can test that you get the `NamedRange` form back.